components/script/dom/htmlelement.rs

```diff
@@ -421,11 +424,68 @@ impl HTMLElementMethods for HTMLElement {
     }
 
     // https://html.spec.whatwg.org/multipage/#the-innertext-idl-attribute
-    fn SetInnerText(&self, _: DOMString) {
-        // XXX (ferjm) implement this.
+    fn SetInnerText(&self, input: DOMString) {
+        // Step 1.
+        let document = document_from_node(self);
+
+        // Step 2.
+        let fragment = DocumentFragment::new(&document);
+
+        // Step 3. The given value is already named 'input'.
+
+        // Step 4.
+        let mut position = input.chars().peekable();
+
+        // Step 5.
+        let mut text = String::new();
+
+        // Step 6.
+        loop {
+            match position.next() {
```

Comment:
nit: Not sure if `while let Some(..)  = ` would help making this code a bit less nested, your call :)